wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsnext/WorkspaceNextApplier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.wsnext;
+
+import com.google.common.annotations.Beta;
+import java.util.Collection;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalEnvironment;
+import org.eclipse.che.api.workspace.server.wsnext.model.CheService;
+
+/**
+ * Applies Workspace.Next configuration to an internal runtime object that represents workspace
+ * runtime configuration on an infrastructure level.
+ *
```

Comment:
I think it may be helpful to add a note that implementations must be bound with map binder and maybe add an example of binding.